yt_dlp/extractor/pbs.py

```diff
@@ -185,14 +185,15 @@ class PBSIE(InfoExtractor):
 
     _VALID_URL = r'''(?x)https?://
         (?:
-           # Direct video URL
-           (?:{})/(?:(?:vir|port)alplayer|video)/(?P<id>[0-9]+)(?:[?/]|$) |
-           # Article with embedded player (or direct video)
-           (?:www\.)?pbs\.org/(?:[^/]+/){{1,5}}(?P<presumptive_id>[^/]+?)(?:\.html)?/?(?:$|[?\#]) |
-           # Player
-           (?:video|player)\.pbs\.org/(?:widget/)?partnerplayer/(?P<player_id>[^/]+)
+            # Player
+            (?:video|player)\.pbs\.org/(?:widget/)?partnerplayer/(?P<player_id>[^/]+) |
+            # Direct video URL, or article with embedded player
+            (?:%s)/(?:
```

Comment:
If the `%` formatting causes linting errors:
```suggestion
            (?:{})/(?:
```
and revert l.195-6.

Reply:
attempting to apply the changes, i'm faced with an error that appears to deal with line 193

```F524 `.format` call is missing argument(s) for placeholder(s): 1,5```

```python
(?:[^/]+/){1,5}(?P<presumptive_id>[^/]+?)(?:\.html)?/?(?:$|[?\#])
```

I'm new to python, so i'm assuming that  `.format()` uses `{}` for replacement/interpolation, but the regex is trying to check for 1 - 5 `/` characters using `{1,5}`?

Any suggestions on how to address this?

Reply:
Yes, originally you had to have something inside the braces, so `{0}` picked up the 0th argument of `format()`, while `{foo}` got the value of keyword argument `foo`. Then empty braces were implicitly interpreted as `{0}`, `{1}`, ... in order of occurrence. Finally a special breaking syntax was added where `f'{foo}'` means `'{}'.format(foo)`. That's the position for the Python versions supported by _yt-dlp_. The `join()` expression that's being spliced in seems to me to be too complex to use the f-string syntax.

Unescaped regex braces in a format string are an actual bug that the linter may not flag, but the resolution is like [in the issue](https://github.com/yt-dlp/yt-dlp/issues/11514#issuecomment-2504357681) (which I ought to have linked): double the braces.

Reply:
Thanks for the explanation! that makes sense! 
Escaping `{1,5}` with `{{1,5}}` allows the linting to pass. Testing was successful too!

I'll keep the update at the ready in case it's needed.

Reply:
After some thought, I've decided to apply the change anyway. It reduces the number of changed lines and keeps the code clearer for anyone else that comes in later to make changes (imo).